orttraining/orttraining/python/training/ortmodule/_io.py

```diff
@@ -91,26 +91,35 @@ def _combine_input_buffers_initializers(param_names, onnx_input_names, input_inf
         if name in kwargs and kwargs[name] is not None:
             # Only use keywords coming from user that are expected by ONNX model
             inp = kwargs[name]
-        elif input_idx < len(non_none_inputs):
-            # Only use positionals coming from user that are expected by ONNX model
-            if name != input_info.names[input_idx]:
-                # When ONNX drops unused inputs, get correct index from user input
-                input_idx = input_info.names.index(name)
-            inp = non_none_inputs[input_idx]
-
-        elif input_idx >= len(non_none_inputs):
+
+        if inp is None:
```

Comment:
In order to use an ```elif``` condition, we will need to change ```input_info.names``` to a map from ```{names->index}```. Otherwise there is no good condition for this check.